osbs/api.py

```diff
@@ -413,10 +413,29 @@ def create_prod_build(self, git_uri, git_ref,
         build_request = self.get_build_request()
         labels = utils.Labels(df_parser.labels)
 
+        required_missing = False
         try:
             _, name_value = labels.get_name_and_value(utils.Labels.LABEL_TYPE_NAME)
+        except KeyError:
```

Comment:
@vrutkovs I think that would still raise an exception on the first miss and not continue with the checks as we want.

How about something a little more dynamic:

```python
labels = {}
required_missing = False
for label in [utils.Labels.LABEL_TYPE_NAME, 
              utils.Labels.LABEL_TYPE_COMPONENT, 
              utils.Labels.LABEL_TYPE_VERSION]:
    try:
        labels[label] = labels.get_name_and_value(label)
    except KeyError:
        required_missing = True
        logger.error("required label missing from Dockerfile : %s",
                     labels.get_name(utils.Labels.LABEL_TYPE_COMPONENT))

if required_missing:
    raise OsbsValidationException("required label(s) missing from Dockerfile")

# ... later ...
build_request.set_params(
    name_label=labels[utils.Labels.LABEL_TYPE_NAME],
    component=labels[utils.Labels.LABEL_TYPE_COMPONENT],
```